StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheetFormFactory/PaymentSheetFormFactory+FormSpec.swift

```diff
@@ -78,8 +153,88 @@ extension PaymentSheetFormFactory {
             return makeIban(apiPath: spec.apiPath?["v1"])
         case .sepa_mandate:
             return makeSepaMandate()
+        case .placeholder(let spec):
+            return makePlaceholder(for: spec)
         case .unknown:
             return nil
         }
     }
+
+    func makePlaceholder(for spec: FormSpec.PlaceholderSpec) -> Element? {
+        let field = spec.field
+        guard field != .unknown else { return nil }
+        return makeOptionalBillingDetailsField(for: field)
+    }
+
+    func makeOptionalBillingDetailsField(for field: FormSpec.PlaceholderSpec.PlaceholderField) -> Element? {
+        switch field {
+        case .name:
+            return configuration.billingDetailsCollectionConfiguration.name == .always ? makeName() : nil
+        case .email:
+            return configuration.billingDetailsCollectionConfiguration.email == .always ? makeEmail() : nil
+        case .phone:
+            return configuration.billingDetailsCollectionConfiguration.phone == .always ? makePhone() : nil
+        case .billingAddress:
+            return configuration.billingDetailsCollectionConfiguration.address == .full
+                ? makeBillingAddressSection(countries: nil)
+                : nil
+        case .billingAddressWithoutCountry:
+            return configuration.billingDetailsCollectionConfiguration.address == .full
+                ? makeBillingAddressSection(collectionMode: .noCountry, countries: nil)
+                : nil
+        case .unknown: return nil
+        }
+    }
+
+    private func connectBillingDetailsFields(
+        countryElement: Element?,
+        addressElement: Element?,
+        phoneElement: Element?
+    ) {
+        // Using a closure because a function would require capturing self, which will be deallocated by the time
+        // the closures below are called.
```

Comment:
I don't follow. If it's capturing self, wouldn't that mean it's not deallocated when the closures are called?

Reply:
`self` should be captured weakly to avoid a retain cycle, and since it is not retained strongly it gets deallocated.
That's how I implemented it first and it wasn't working until I realized that was the issue, which is why I added the comment.

Reply:
For my benefit, where's the retain cycle? Is the `PaymentSheetFormFactory` holding a strong reference to the element somewhere?

Reply:
No, the closures on line 213 and 231 would need to capture `self` in order to call a function on it, that way the closure would hold a reference to `self` and `self` holds a reference to the closure, it's the most classic example of a retain cycle, so you normally capture `self` weakly to avoid it, but in this case the factory will be deallocated before the closures are called.
I also thought about capturing a reference to the function, but IMO it looks clearer if you use an inline closure an capture that, which is basically the same effect.

Reply:
> IMO it looks clearer if you use an inline closure an capture that, which is basically the same effect.

I agree that the closure looks cleaner either way, so just having a theoretical conversation at this point to improve my understanding of our code, but: `self` _doesn't_ hold a reference to the closure since it's being set on a property of the element, and I don't see where `self` has a strong reference to the element. So I'm not understanding where the retain cycle would come from here?

Reply:
Here is the memory leak 
<img width="1469" alt="Screen Shot 2023-03-11 at 11 56 52 AM" src="https://user-images.githubusercontent.com/114543737/224509023-d40748f4-d741-49f5-ac0c-e751c5e0b609.png">

It is an indirect cycle in this case, or maybe not a cycle but it causes a memory leak since the factory would be retained far longer than it should.